feature/gribi/otg_tests/route_removal_non_default_vrf_test/route_removal_non_default_vrf_test.go

```diff
@@ -0,0 +1,441 @@
+	}
+}
+
+// configureDUT configures port1-2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	d := gnmi.OC()
+
+	p1 := dut.Port(t, "port1")
+	p2 := dut.Port(t, "port2")
+
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), dutPort1.NewOCInterface(p1.Name()))
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), dutPort2.NewOCInterface(p2.Name()))
+
+	if *deviations.ExplicitPortSpeed {
+		fptest.SetPortSpeed(t, p1)
+		fptest.SetPortSpeed(t, p2)
+	}
+	if *deviations.ExplicitInterfaceInDefaultVRF {
+		fptest.AssignToNetworkInstance(t, dut, p2.Name(), *deviations.DefaultNetworkInstance, 0)
+	}
+
+}
+
+// configureATE configures port1, port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) gosnappi.Config {
+	t.Helper()
+	top := ate.OTG().NewConfig(t)
+
+	p1 := ate.Port(t, "port1")
+	p2 := ate.Port(t, "port2")
+
+	atePort1.AddToOTG(top, p1, &dutPort1)
+	atePort2.AddToOTG(top, p2, &dutPort2)
+
+	// Adding traffic flow
+	flowipv4 := top.Flows().Add().SetName("Flow")
+	flowipv4.Metrics().SetEnable(true)
+	e1 := flowipv4.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	flowipv4.TxRx().Device().SetTxNames([]string{atePort1.Name + ".IPv4"}).SetRxNames([]string{atePort2.Name + ".IPv4"})
+	v4 := flowipv4.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().Increment().SetStart("198.51.100.0").SetCount(255)
+
+	return top
+}
+
+// configureNetworkInstance configures network instance.
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+
+	nonDefaultNI := networkInstance(t, nonDefaultVRF)
+	p1 := dut.Port(t, "port1")
+	niIntf := nonDefaultNI.GetOrCreateInterface(p1.Name())
+	niIntf.Subinterface = ygot.Uint32(0)
+	niIntf.Interface = ygot.String(p1.Name())
+
+	gnmi.Replace(t, dut, gnmi.OC().NetworkInstance(nonDefaultVRF).Config(), nonDefaultNI)
+}
+
+// networkInstance creates an OpenConfig network instance with the specified name
+func networkInstance(t *testing.T, name string) *oc.NetworkInstance {
+	d := &oc.Root{}
+	ni := d.GetOrCreateNetworkInstance(name)
+	ni.Description = ygot.String("Non Default routing instance created for testing")
+	ni.Type = oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L3VRF
+	return ni
+}
+
+// testTraffic generates traffic flow from source network to
+// destination network via srcEndPoint to dstEndPoint and checks for
+// packet loss.
+func testTraffic(t *testing.T, ate *ondatra.ATEDevice, config gosnappi.Config) int64 {
+	ate.OTG().StartTraffic(t)
+	time.Sleep(15 * time.Second)
+	ate.OTG().StopTraffic(t)
+	otgutils.LogFlowMetrics(t, ate.OTG(), config)
+
+	flowMetric := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow("Flow").State())
+	txPackets := flowMetric.GetCounters().GetOutPkts()
+	rxPackets := flowMetric.GetCounters().GetInPkts()
+	lossPct := int64((txPackets - rxPackets) * 100 / txPackets)
```

Comment:
Pardon me, I noticed one more thing.

Can you add a separate check for txPackets being zero? Then we can have a clearer error message in case of division by zero.

Reply:
Also, I'd like for the lossPct checks at the callsites to be clearer... It's not clear what the return value of "testTraffic" is unless we go look at the function.

I'm assuming you changed this as using the LossPct() telemetry doesn't work (if that does work, that would be a good solution)

Could you split the lossPct calculation into a separate test helper with something like "lossPct" in the name?

Reply:
I've added the check for 0 txPackets and renamed the call to **getTrafficPacketLoss**

Reply:
Hm, "get" makes it sound like it just takes a GNMI Get (or maybe a little bit more work) to get the packet loss. We also generate traffic and block for 15 seconds...

I think a slightly different name that makes it sound like more work is done such as computeTrafficPacketLoss would be okay.

See also guidance at https://google.github.io/styleguide/go/decisions#getters

As another option, it may be good to split into two functions so we can have clear names for both steps. Something like:

func generateTraffic(t *testing.T, ate *ondatra.ATEDevice, config gosnappi.Config) {
	t.Helper()
	ate.OTG().StartTraffic(t)
	time.Sleep(15 * time.Second)
	ate.OTG().StopTraffic(t)
	otgutils.LogFlowMetrics(t, ate.OTG(), config)
}

func getTrafficPacketLoss(t *testing.T, ate *ondatra.ATEDevice) int64 {
	t.Helper()
	ate.OTG().StartTraffic(t)
	time.Sleep(15 * time.Second)
	ate.OTG().StopTraffic(t)
	otgutils.LogFlowMetrics(t, ate.OTG(), config)

	flowMetric := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow("Flow").State())
	txPackets := flowMetric.GetCounters().GetOutPkts()
	if txPackets == 0 {
		t.Fatal("Got 0 txPackets. Want >0")
	}
	...
}

Reply:
Also, whether splitting the function are not, I think these should be marked as test helpers with t.Helper(). See the example in the prior comment.

Also, t.Fatal is better here so we don't go forward testing with an invalid "0" return value.

See guidance on test helpers at https://google.github.io/styleguide/go/decisions#test-helpers

Reply:
Split the call into 2 functions and renamed to computeLossPct. Added t.Fatal in case of 0 tx packets